Banyan/src/requests.jl

```diff
@@ -366,11 +366,13 @@ function send_evaluation(value_id::ValueId, job_id::JobId)
     global encourage_parallelism_with_batches
     global exaggurate_size
 
+    # First, ensure that the job is running
+    wait_for_job(job_id)
+
```

Comment:
Isn't `wait_for_job` somewhat expensive? Shouldn't we only do this if we haven't already waited for this job?

Reply:
Hmm, that's a good point. We don't know whether we've already waited for a job, but we can cache the job status, especially useful if we know status is "running".
We can also do what you suggested, where we call `evaluate` and see if it fails first

Reply:
Now, we allow `evaluate` to run, as long as the job is either creating or running. However, the client side cannot read the queue until the job is running.